cltk/inflection/old_norse/phonemic_rules.py

```diff
@@ -0,0 +1,391 @@
+"""Old Norse phonemic rules
+
+Old Norse words encounter many sound changes which occur mainly according to endings.
+Some sound changes may be explained by disappeared or transformed endings.
+
+The i ending lead to an i-umlaut (sound transformation) which made the previous vowel more alike the sound i.
+The u ending lead to an u-umlaut (sound transformation) which made the previous vowel more alike the sound u.
+The r ending is sometimes assimilated depending on the previous adjacent consonant.
+
+These sound transformations were active at an earlier stage of "classical" Old Norse, i.e. conditions in which such
+transformations occurred may be not encountered in "classical" Old Norse, nevertheless their results were still visible
+in "classical" Old Norse.
+
+Sources:
+- Kleine Grammatik des Altisländischen by Robert Nedoma (in German)
+- https://en.wikipedia.org/wiki/Proto-Norse_language#Proto-Norse_to_Old_Norse
+
+"""
+
+from cltk.phonology.syllabify import Syllabifier, Syllable
+from cltk.corpus.old_norse.syllabifier import invalid_onsets, VOWELS, CONSONANTS, SHORT_VOWELS, LONG_VOWELS, \
+    DIPHTHONGS, BACK_TO_FRONT_VOWELS
+
+import numpy
+
+__author__ = ["Clément Besnier <clemsciences@aol.com>", ]
+
+s = Syllabifier(language="old_norse", break_geminants=True)
+s.set_invalid_onsets(invalid_onsets)
+
+
+class OldNorseSyllable(Syllable):
+
+    def apply_u_umlaut(self, is_second=False):
+        if "".join(self.nucleus) == "a":
+            self.nucleus = ["ö"]
+        elif "".join(self.nucleus) == "ö" and is_second:
+            self.nucleus = ["u"]
+
+
+def extract_common_stem(*args):
```

Comment:
I really like this idea and I could experiment with something hopefully more accurate. What I have in mind:

- Strip prefixes and suffixes (if the option is offered)
- Fairly straight-forward brute-force: Keep a stack of possible phonetic replacements and keep updating both of the strings, keeping track of the longest substring.

This doesn't account for inflections (which is half the point anyway), so this should be preferably coupled with stemming.

Reply:
The idea I had was to align the three forms with each other and see from which form it is the cheapest to transform to the two other forms, if we consider addition, deletion and modification as operations which cost.